woocommerce/woocommerce
# Revert PR#39146: fix basic flat rate shipping scenario

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





- Reverts https://github.com/woocommerce/woocommerce/pull/39146 (note: original PR was https://github.com/woocommerce/woocommerce/pull/37847).
- This is the basis of a nice change, but unfortunately it seems we missed a few aspects during review and it will break some common shipping scenarios.
- I have re-opened original issue https://github.com/woocommerce/woocommerce/issues/27030 so we can pick this back up at some future point (and, when we do, we should add additional tests to cover the scenario described here).



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Start by using the [code currently in `trunk`](https://github.com/woocommerce/woocommerce/tree/8a3893c47c238010f22d397bc30a4b4c404254be). 

1. Create a single shipping zone with a single flat rate shipping method. The shipping method should have a cost, but should not have any class-specific costs. For example, you could setup:
    a. Shipping zone: Canada
    b. Flat rate shipping method: cost $5

<table><tr><td><img width="600" alt="Zone settings" src="https://github.com/woocommerce/woocommerce/assets/3594411/17ddcf47-7736-4664-95ad-6c98b749327f"></td><td>
<img width="600" alt="Flat rate settings" src="https://github.com/woocommerce/woocommerce/assets/3594411/6e89a31b-e356-47bf-8416-96233f99fd3f"></td></tr></table>

2. Define a shipping class, name it whatever you like.
3. As a shopper, add a product (without a shipping class) to the cart, proceed to checkout, and enter an address that will match the defined shipping zone. To build on my previous example, this might be:

> 1001 Testgood Lane
> Fort Quality
> British Columbia
> V0E 1S1
> Canada

4. Notice that no shipping methods are available for selection and you cannot checkout. With this branch, however, your flat rate shipping fee of $5 should be available.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
